src/main/java/Duke.java

```diff
@@ -1,10 +1,264 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+    private static final String INPUT_PROMPT = "$ ";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_BYE = "bye";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+
+    private static final String TASK_DEADLINE_SPLITTER = "/by";
+    private static final String TASK_EVENT_SPLITTER = "/at";
+
+    private static final String MESSAGE_WALL = "------------------------------------------------------------";
+    private static final String MESSAGE_LIST_EMPTY = "List is empty";
+    private static final String MESSAGE_LIST_HEADER = "Task List:";
+    private static final String MESSAGE_INVALID_TASK_NUMBER = "Invalid task number";
+    private static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
+    private static final String MESSAGE_WELCOME = "Hello! I'm Duke\nWhat can I do for you?";
+    private static final String MESSAGE_BYE = "Bye. Hope to see you again soon!";
+
+    private static final String MESSAGE_FORMAT_DONE_USAGE = "Usage: %s <task number>";
+    private static final String MESSAGE_FORMAT_TASK_ALREADY_MARKED = "Task #%d is already marked as done";
+    private static final String MESSAGE_FORMAT_TASK_MARKED = "Task marked as done:\n  %s";
+    private static final String MESSAGE_FORMAT_DEADLINE_USAGE = "Usage: %s <description> %s <date/time>";
+    private static final String MESSAGE_FORMAT_EVENT_USAGE = "Usage: %s <description> %s <date/time>";
+    private static final String MESSAGE_FORMAT_TASK_ADDED = "Got it. Task added:\n%s\nThere are %d tasks in the list";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static final ArrayList<Task> TASKS = new ArrayList<>();
```

Comment:
Nice that you used the plural form of "TASK" since you are representing a collection of Tasks.